src/components/Auth/ResetPassword.tsx

```diff
@@ -0,0 +1,157 @@
+      duration: 3.5,
+      // close model too
+    });
+  };
+
+  const handlePasswordReset = useCallback(() => {
+    if (password === confirmPassword) {
+      onPasswordReset({ login, password, token });
+    }
+  }, [login, token, password, confirmPassword, onPasswordReset]);
+
+  async function validatePassword(password: string): Promise<void> {
+    setEnableSubmitButton(false);
+    if (!/(?=.*[A-Z])(?=.*[a-z]).*/.test(password)) {
+      return Promise.reject(new Error('Password must contain lower and uppercase characters'));
+    }
+    if (!/.{8,}/.test(password)) {
+      return Promise.reject(new Error('Password must be at least 8 characters'));
+    }
+    if (!/(?=.*[0-9\W]).*/.test(password)) {
+      return Promise.reject(new Error('Password must contain a number or a symbol'));
+    }
+    setEnableSubmitButton(true);
+  }
+  const checkPasswordMatch = useCallback(() => {
+    if (!(password === confirmPassword)) {
+      return Promise.reject(new Error('Passwords must match'));
+    }
+  }, [password, confirmPassword]);
+
+  useEffect(() => {
+    if (isPasswordResetted) {
+      resetSuccessfulNotification();
+      history.push('/login/');
+    }
+  }, [history, isPasswordResetted]);
+
+  return (
+    <Container>
+      <Form layout="vertical">
+        <Typography.Title level={3} className="title">
+          {`Reset Password for ${login.replace(/\s/g, '')}`}
+        </Typography.Title>
+        <Form.Item
+          name="password"
+          rules={[
+            { required: true, message: 'Password is required' },
+            { validator: (_, value) => validatePassword(value) },
+            // { validator: () => checkPasswordMatch() },
+          ]}
+        >
+          <Input
+            type="password"
+            size="large"
+            style={{ width: '240px' }}
+            placeholder="New Password"
+            disabled={authenticating}
+            value={password}
+            required={true}
+            onChange={handlePasswordChange}
+          />
+        </Form.Item>
+        <Form.Item
+          name="confirmPassword"
+          rules={[
+            { required: true, message: 'Confirm password is required' },
+            { validator: (_, value) => validatePassword(value) },
+            { validator: () => checkPasswordMatch() },
+          ]}
+        >
+          <Input
+            type="password"
+            size="large"
+            style={{ width: '240px' }}
+            placeholder="Confirm New Password"
+            disabled={authenticating}
+            value={confirmPassword}
+            required={true}
+            onChange={handleconfirmPasswordChange}
+          />
+        </Form.Item>
+        <Button
+          type="primary"
+          size="large"
+          htmlType="submit"
+          loading={authenticating}
+          disabled={!(password === confirmPassword) || !enableSubmitButton}
```

Comment:
no need for this type of logic. in Antd the preferred way is to just keep the submit buttons enabled and just run validation when clicked. just use `authenticating` for disabling and run validation otherwise. you can check https://github.com/chronologic/automate-ui/blob/main/src/components/Strategies/StrategyDetails.tsx#L90 (and the whole function as well as the rest of the file) for an example